### PR TITLE
SW-4961 Refactor the deliverables table with search and filters into its own component so we can easily resuse for people table

### PR DESCRIPTION
**Pull request by nickgraz**

### CODE REVIEW

**Comment by nickgraz** [comment]
* **#2312** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2312?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2311** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2311?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/TableWithSearchFilters/index.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            search={searchValue}
+            onSearch={setSearchValue}
+            currentFilters={filters}
+            setCurrentFilters={setFilters}
+            featuredFilters={_featuredFilters}
+          />
+        </Grid>
+
+        <Grid item xs={12}>
+          <Table
```
Comment (karthikbtf) [inline src/components/TableWithSearchFilters/index.tsx:131]: Use the column order preserved table, with unique ids?

File: src/components/TableWithSearchFilters/index.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import React, { RefObject, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { Container, Grid } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { RendererProps, SortOrder, TableColumnType, TableRowType } from '@terraware/web-components';
+
+import Card from 'src/components/common/Card';
+import SearchFiltersWrapperV2, { FilterConfig } from 'src/components/common/SearchFiltersWrapperV2';
+import { BaseTable as Table } from 'src/components/common/table';
```
Comment (karthikbtf) [inline src/components/TableWithSearchFilters/index.tsx:9]: The order preserved table allows us to shuffle columns around.

**Review by karthikbtf — APPROVED** [review]
looks good, one suggestion on table

**Review by karthikbtf — APPROVED** [review]
Thank you 🙏 
Chudo's  request/requirement that all tables support column shuffling with persistence.

**Comment by nickgraz** [comment]
No problem! Sorry you've had to remind me a few times now

**Pull request merged**